funflow/src/Funflow/Config.hs

```diff
@@ -26,13 +27,14 @@ type ConfigKey = Text
 -- initially being read in order to identify which config keys are defined in
 -- the file and therefore is of type Object.
 
+-- this has changed to be a KeyMap Value in the latest version of yaml
 type ConfigMap = Object
 
-type EnvConfigMap = HashMap.HashMap Text String
+type EnvConfigMap = Ae.KeyMap String
 
 data ExternalConfig = ExternalConfig
-  { fileConfig :: ConfigMap,
-    envConfig :: EnvConfigMap
+  { fileConfig :: !ConfigMap,
```

Comment:
I started doing this just because it's considered safer to have record fields as strict (when they'll definitely be used), but I soon realised that there are _many_ record types and it would be a huge change. I'm happy to revert this to keep things related to the actual change.

Reply:
Having said that, it might be worth updating the strictness (and making some other best-practice changes) in a separate PR.